Only configure server-side spring components when running in servlet container

I am trying to use belgif-rest-problem-spring in a Spring Shell application.
This is a standalone CLI application that does not run in a servlet container. It only requires the client-side integrations of the problem library, and not the server-side integrations.

But currently, components like https://github.com/belgif/rest-problem-java/blob/main/belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/RoutingExceptionsHandler.java are registered and they require jakarta EE servlet libraries that are not on the classpath.

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'routingExceptionsHandler': Lookup method resolution failed
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.checkLookupMethods(AutowiredAnnotationBeanPostProcessor.java:498) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.determineCandidateConstructors(AutowiredAnnotationBeanPostProcessor.java:368) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineConstructorsFromBeanPostProcessors(AbstractAutowireCapableBeanFactory.java:1314) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1209) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:562) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:975) ~[spring-beans-6.1.12.jar:6.1.12]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:971) ~[spring-context-6.1.12.jar:6.1.12]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:625) ~[spring-context-6.1.12.jar:6.1.12]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-3.3.3.jar:3.3.3]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456) ~[spring-boot-3.3.3.jar:3.3.3]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:335) ~[spring-boot-3.3.3.jar:3.3.3]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1363) ~[spring-boot-3.3.3.jar:3.3.3]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1352) ~[spring-boot-3.3.3.jar:3.3.3]
	at be.fgov.kszbcss.batch.operator.cli.BatchOperatorCLI.main(BatchOperatorCLI.java:12) ~[classes/:na]
Caused by: java.lang.IllegalStateException: Failed to introspect Class [io.github.belgif.rest.problem.spring.RoutingExceptionsHandler] from ClassLoader [jdk.internal.loader.ClassLoaders$AppClassLoader@76ed5528]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:483) ~[spring-core-6.1.12.jar:6.1.12]
	at org.springframework.util.ReflectionUtils.doWithLocalMethods(ReflectionUtils.java:320) ~[spring-core-6.1.12.jar:6.1.12]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.checkLookupMethods(AutowiredAnnotationBeanPostProcessor.java:476) ~[spring-beans-6.1.12.jar:6.1.12]
	... 18 common frames omitted
Caused by: java.lang.NoClassDefFoundError: jakarta/servlet/ServletException
```

I guess this can be solved by adding some @ConditionalOnClass annotations?

--- a/belgif-rest-problem-spring-boot-3/src/main/java/io/github/belgif/rest/problem/spring/NoResourceFoundExceptionHandler.java
+++ b/belgif-rest-problem-spring-boot-3/src/main/java/io/github/belgif/rest/problem/spring/NoResourceFoundExceptionHandler.java
@@ -1,5 +1,6 @@
 package io.github.belgif.rest.problem.spring;
 
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
 import org.springframework.core.annotation.Order;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.ExceptionHandler;
@@ -11,6 +12,7 @@ import io.github.belgif.rest.problem.api.Problem;
 
 @RestControllerAdvice
 @Order(1)
+@ConditionalOnWebApplication
 public class NoResourceFoundExceptionHandler {
 
     @ExceptionHandler(NoResourceFoundException.class)

--- a/belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/BeanValidationExceptionsHandler.java
+++ b/belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/BeanValidationExceptionsHandler.java
@@ -7,6 +7,7 @@ import java.util.stream.Collectors;
 import javax.validation.ConstraintViolationException;
 
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
 import org.springframework.core.annotation.Order;
 import org.springframework.http.ResponseEntity;
 import org.springframework.validation.BindException;
@@ -30,6 +31,7 @@ import io.github.belgif.rest.problem.internal.DetermineSourceUtil;
 @RestControllerAdvice
 @Order(1)
 @ConditionalOnClass(ConstraintViolationException.class)
+@ConditionalOnWebApplication
 public class BeanValidationExceptionsHandler {
 
     @ExceptionHandler(ConstraintViolationException.class)

--- a/belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/ProblemExceptionHandler.java
+++ b/belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/ProblemExceptionHandler.java
@@ -2,6 +2,7 @@ package io.github.belgif.rest.problem.spring;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
@@ -18,6 +19,7 @@ import io.github.belgif.rest.problem.api.Problem;
  * </p>
  */
 @RestControllerAdvice
+@ConditionalOnWebApplication
 public class ProblemExceptionHandler {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ProblemExceptionHandler.class);

--- a/belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/RoutingExceptionsHandler.java
+++ b/belgif-rest-problem-spring/src/main/java/io/github/belgif/rest/problem/spring/RoutingExceptionsHandler.java
@@ -1,5 +1,6 @@
 package io.github.belgif.rest.problem.spring;
 
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
 import org.springframework.core.annotation.Order;
 import org.springframework.http.ResponseEntity;
 import org.springframework.http.converter.HttpMessageNotReadableException;
@@ -18,6 +19,7 @@ import io.github.belgif.rest.problem.api.Problem;
  */
 @RestControllerAdvice
 @Order(1)
+@ConditionalOnWebApplication
 public class RoutingExceptionsHandler {
 
     @ExceptionHandler(MissingServletRequestParameterException.class)

--- a/src/main/asciidoc/index.adoc
+++ b/src/main/asciidoc/index.adoc
@@ -35,6 +35,10 @@ The library consists of these modules:
 
 * Fix validation for "overflow" SSINs
 
+*belgif-rest-problem-spring:*
+
+* Make RestControllerAdvice components @ConditionalOnWebApplication
+
 === Version 0.5
 
 *belgif-rest-problem-spring:*

